respy/shared.py

```diff
@@ -501,3 +501,21 @@ def calculate_expected_value_functions(
         expected_value_functions[0] += max_value_functions
 
     expected_value_functions[0] /= n_draws
+
+
+def convert_dictionary_keys_to_dense_indices(dictionary):
```

Comment:
It may happen that the values of the dense variables are floats and not ints. I think in the simulation where the DataFrame contains NaNs. In this case we need to convert to integers.